File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/file_group/file/FileEntity.java
package pl.madzierski.daniel.app.file_group.file;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import pl.madzierski.daniel.app.common.model.BaseEntity;
import pl.madzierski.daniel.app.file_group.FileGroupEntity;

@Entity
@Table(name = "receipt_file")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class FileEntity extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "file_group_id")
    private FileGroupEntity fileGroup;

    private String path;

    @Column(name = "raw_data", columnDefinition = "TEXT")
    private String rawData;

    private Integer partNumber;
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/file_group/file/FileEntity.kt
package pl.madzierski.daniel.app.file_group.file

import jakarta.persistence.*
import pl.madzierski.daniel.app.common.model.BaseEntity
import pl.madzierski.daniel.app.file_group.FileGroupEntity
import pl.madzierski.daniel.app.file_group.FileType
import pl.madzierski.daniel.app.receipt.ReceiptEntity

@Entity
@Table(name = "receipt_file")
data class FileEntity(

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "file_group_id", nullable = true)
    val fileGroup: FileGroupEntity,

    var path: String? = null,

    @Column(name = "raw_data", columnDefinition = "TEXT")
    var rawData: String? = null,

    val partNumber: Int?

) : BaseEntity() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FileEntity

        if (partNumber != other.partNumber) return false
        if (path != other.path) return false
        if (rawData != other.rawData) return false

        return true
    }

    override fun hashCode(): Int {
        var result = partNumber ?: 0
        result = 31 * result + (path?.hashCode() ?: 0)
        result = 31 * result + (rawData?.hashCode() ?: 0)
        return result
    }

}
